### PR TITLE
Add NoOpQuantizer for use with Bop

### PR DESCRIPTION
**Pull request by lgeiger**
This implements the last missing change outlined in #306

### CODE REVIEW

File: larq/optimizers.py [hunk 1]
```diff
@@ -296,11 +302,11 @@ def from_config(cls, config, custom_objects=None):
...
-        """Returns True for binary variables named using the Larq Zoo naming scheme.
+        """Returns `True` for variables with `var.precision == 1`.
 
         This is an example of a predictate that can be used by the `CaseOptimizer`.
 
         # Arguments
         var: a `tf.Variable`.
         """
-        return "/kernel" in var.name and "quant_" in var.name
+        return getattr(var, "precision", 32) == 1
```
Comment (leonoverweel) [inline larq/optimizers.py]: 😍

File: larq/optimizers_test.py [hunk 2]
```diff
@@ -109,7 +109,12 @@ def test_weights(self):
         model = tf.keras.Sequential(
             [
                 tf.keras.layers.Flatten(input_shape=(28, 28)),
-                lq.layers.QuantDense(64, input_quantizer="ste_sign", activation="relu"),
+                lq.layers.QuantDense(
+                    64,
+                    input_quantizer="ste_sign",
+                    kernel_quantizer=lq.quantizers.NoOpQuantizer(1),
```
Comment (leonoverweel) [inline larq/optimizers_test.py]:
```suggestion
                    kernel_quantizer=lq.quantizers.NoOpQuantizer(precision=1),
```

File: larq/quantizers.py [hunk 3]
```diff
@@ -164,6 +165,45 @@ def non_trainable_weights(self):
         return []
 
 
+@utils.register_keras_custom_object
+class NoOpQuantizer(BaseQuantizer):
+    r"""Instantiates a serializable no-op quantizer.
```
Comment (leonoverweel) [inline larq/quantizers.py]: Why the `r`? You're already escaping `\`s right?

Comment (lgeiger) [reply larq/quantizers.py]: I copied this from the other quantizers, so I guess it is needed to correctly render MatJax.

**Comment by lgeiger** [comment]
> Maybe it'd also be good to add some documentation explaining the purpose of the NoOpQuantizer for optimizers like Bop.

I added more documentation in the last commit.

**Pull request merged**